### PR TITLE
Add TryGetLinearlySeparableComponents and tests

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This is all @Sergio0694 brilliant work from [sp/auto-matrix-linear-decomposition](https://github.com/SixLabors/ImageSharp/tree/bdf1b05a12ed6def547df9c3b6c755344ceab58e). All I've done is copy the code and tests from that branch as it's impossible to merge it following the Git LFS changes.

### CODE REVIEW

**Comment by Sergio0694** [comment]
Ooh I forgot about this branch, really happy to see it back!
And glad to see that my work with that linear separation method will be useful! 😄

**Comment by JimBobSquarePants** [comment]
Very useful!!

I'll be incorporating it into the general convolution API I'll be exposing.

File: src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessorHelpers.cs [hunk 1]
```diff
@@ -83,4 +88,99 @@ internal static float[] CreateGaussianSharpenKernel(int size, float weight)
...
+        //   2) It checks that every j-th item in the row respects C[i] = M[i, j] / M[0, j]. If this is
+        //      not true for any j-th item in any i-th row, then the matrix is not linearly separable.
+        //   3) It sets items in R and C to the values detailed above if the validation passed.
+        for (int y = 1; y < height; y++)
+        {
+            float ratio = matrix[y, 0] / matrix[0, 0];
+
+            for (int x = 1; x < width; x++)
+            {
+                if (Math.Abs(ratio - (matrix[y, x] / matrix[0, x])) > 0.0001f)
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessorHelpers.cs:159]: Using hardcoded epsilon might be problematic in many cases. Personally I prefer to make it a parameter with a default value or at least to work with library-wide constants proven to work well for the given library.

What are ImageSharp use-cases for this feature?

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessorHelpers.cs:159]: Yeah we should normalise constants. There might be one in Numerics already.

I plan to use this as an underlying optimisation when exposing a general convolution API.

**Pull request merged**